### ISSUE TITLE
8.40 Generated suppression filter file is unparseable

### ISSUE THREAD

**Issue by alanbuttars** [checkstyle/checkstyle#9244]
This bug appears in version *8.40*

I configure my project with gradle:

_build.gradle_
```gradle
apply plugin: 'checkstyle'

checkstyle {
    toolVersion = '8.40'
}
```

_config/checkstyle/checkstyle.xml_
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
    
<module name = "Checker">
    <module name="SuppressionFilter">
        <property name="file" value="config/checkstyle/suppressions.xml"/>
    </module>
    ...many other lines
</module>
```

There are plenty of warnings in my project, so I generated a`config/checkstyle/suppression.xml` file with the command line tool:
```
java -jar ~/Downloads/checkstyle-8.40-all.jar com.puppycrawl.tools.checkstyle.Main -c config/checkstyle/checkstyle.xml -g -o config/checkstyle/suppressions.xml . 
```

This succeeds in generating the file. But when I try to run `./gradlew clean build -x test`, my `suppressions.xml` file is unparseable:

```
* What went wrong:
Execution failed for task ':checkstyleMain'.
> java.lang.IllegalArgumentException: Unexpected xpath query: /CLASS_DEF[./IDENT[@text='MyService']]/OBJBLOCK/METHOD_DEF[./IDENT[@text='onAction']]/SLIST/EXPR/METHOD_CALL[./IDENT[@text='ifCondition']]/ELIST/LAMBDA/SLIST/LITERAL_IF/SLIST/EXPR/METHOD_CALL/ELIST/EXPR/PLUS[./STRING_LITERAL[@text='Partially completed task id={}, transactionId={} ']]
```

If I comment out the offending rule, I can proceed, BUT my entire suppressions.xml file gets ignored.

**Comment by rnveach** [comment]
@alanbuttars  please read over https://checkstyle.org/report_issue.html#How_to_report_a_bug.3F .

To fix your issue we need to be able to reproduce it so we can identify the problem in our code. We need a minimal Java input file and the module and it's configuration that produces the violation and was used to generate the suppression.

You can remove any excess code from that example as you wish. We just need to reproduce the issue.

**Comment by alanbuttars** [comment]
Got it, sorry about that. I'm unable to recreate the "unparseable suppressions.xml" bug with sample code, but the suppressions.xml is getting completely ignored now: https://github.com/checkstyle/checkstyle/issues/9245
